### PR TITLE
test: add integration tests for Server Info requests

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Add integration tests for Server Info requests

### Type of Change

- [X] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Test Plan
Passes all the tests

### CODE REVIEW

File: test/integration/requests/fee.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  beforeEach(_.partial(setupClient, serverUrl))
+  afterEach(teardownClient)
+
+  it('fee', async function () {
+    const request: FeeRequest = {
+      command: 'fee',
+    }
+    const response = await this.client.request(request)
+    assert.equal(response.status, 'success')
+    assert.equal(response.type, 'response')
```
Comment (mvadari) [inline test/integration/requests/fee.ts:24]: This should also check the shape of the response object and make sure we have that right.

**Review by mvadari — CHANGES_REQUESTED** [review]
This should also check the shape of the response object and make sure we have that right, with the objects that we have.

**Comment by mvadari** [comment]
This PR should be `test:` not `refactor:`

File: test/integration/requests/fee.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import _ from 'lodash'
+
+import { FeeRequest } from '../../../src'
+import serverUrl from '../serverUrl'
+import { setupClient, suiteClientSetup, teardownClient } from '../setup'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('Server Info Fee', function () {
```
Comment (mvadari) [inline test/integration/requests/fee.ts:11]:
```suggestion
describe('fee', function () {
```

File: test/integration/requests/serverInfo.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import _ from 'lodash'
+
+import { ServerInfoRequest } from '../../../src'
+import serverUrl from '../serverUrl'
+import { setupClient, suiteClientSetup, teardownClient } from '../setup'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('Server Info ServerInfo', function () {
```
Comment (mvadari) [inline test/integration/requests/serverInfo.ts:11]:
```suggestion
describe('server_info', function () {
```

File: test/integration/requests/serverState.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import _ from 'lodash'
+
+import { ServerStateRequest } from '../../../src'
+import serverUrl from '../serverUrl'
+import { setupClient, suiteClientSetup, teardownClient } from '../setup'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('Server Info ServerInfo', function () {
```
Comment (mvadari) [inline test/integration/requests/serverState.ts:11]:
```suggestion
describe('server_state', function () {
```

File: test/integration/requests/serverInfo.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  this.timeout(TIMEOUT)
+
+  before(suiteClientSetup)
+  beforeEach(_.partial(setupClient, serverUrl))
+  afterEach(teardownClient)
+
+  it('serverInfo', async function () {
```
Comment (mvadari) [inline test/integration/requests/serverInfo.ts:18]:
```suggestion
  it('base', async function () {
```

File: test/integration/requests/serverState.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  this.timeout(TIMEOUT)
+
+  before(suiteClientSetup)
+  beforeEach(_.partial(setupClient, serverUrl))
+  afterEach(teardownClient)
+
+  it('serverInfo', async function () {
```
Comment (mvadari) [inline test/integration/requests/serverState.ts:18]:
```suggestion
  it('base', async function () {
```

File: test/integration/requests/fee.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  this.timeout(TIMEOUT)
+
+  before(suiteClientSetup)
+  beforeEach(_.partial(setupClient, serverUrl))
+  afterEach(teardownClient)
+
+  it('fee', async function () {
```
Comment (mvadari) [inline test/integration/requests/fee.ts:18]: nit:
```suggestion
  it('base', async function () {
```

**Review by mvadari — CHANGES_REQUESTED** [review]
Check the shape of the response to make sure that it matches what we actually get from rippled. You can see Jackson's PRs for examples.

File: test/integration/requests/fee.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { assert } from 'chai'
+import _ from 'lodash'
+
+import { FeeRequest } from '../../../src'
```
Comment (JST5000) [inline test/integration/requests/fee.ts:4]: This has to import from 'xrpl-local' for the browser test to run

File: test/integration/requests/serverInfo.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { assert } from 'chai'
+import _ from 'lodash'
+
+import { ServerInfoRequest } from '../../../src'
```
Comment (JST5000) [inline test/integration/requests/serverInfo.ts:4]: Ditto

File: test/integration/requests/serverState.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { assert } from 'chai'
+import _ from 'lodash'
+
+import { ServerStateRequest } from '../../../src'
```
Comment (JST5000) [inline test/integration/requests/serverState.ts:4]: Ditto

**Review by JST5000 — CHANGES_REQUESTED** [review]
This test should also be added to the `localIntegrationRunner` so that these tests are part of the browser build - See how the utility test is added with a <script...> tag (Or you can look at this PR to see how I did it: https://github.com/XRPLF/xrpl.js/pull/1636)

**Review by JST5000 — APPROVED** [review]
(Changing to approved so I don't block you :) - Make sure to update the imports before merging though)

File: test/integration/requests/fee.ts [hunk 11]
```diff
@@ -1,26 +1,83 @@
...
-  it('fee', async function () {
+  it('base', async function () {
     const request: FeeRequest = {
       command: 'fee',
     }
     const response = await this.client.request(request)
-    assert.equal(response.status, 'success')
-    assert.equal(response.type, 'response')
+    const expected = {
+      id: 5,
```
Comment (mvadari) [inline test/integration/requests/fee.ts:25]: nit:
```suggestion
      id: 0,
```
To follow what the fixtures do. Apply to the rest of the files as well.

File: test/integration/requests/fee.ts [hunk 12]
```diff
@@ -1,26 +1,83 @@
...
+      _.every(
+        Object.keys(expected.result.drops),
+        // eslint-disable-next-line @typescript-eslint/unbound-method -- need has
+        _.partial(_.has, response.result.drops),
+      ),
+      true,
+    )
+    for (const key of Object.keys(response.result.drops)) {
+      assert.equal(typeof response.result.drops[key], 'string')
+    }
```
Comment (mvadari) [inline test/integration/requests/fee.ts:68]: What are you trying to accomplish here? Honestly, the numbers should be the same, given that everything is using the same docker container - you should just be able to compare them directly.

File: test/integration/requests/fee.ts [hunk 13]
```diff
@@ -1,26 +1,83 @@
...
+      _.every(
+        Object.keys(expected.result.levels),
+        // eslint-disable-next-line @typescript-eslint/unbound-method -- need has
+        _.partial(_.has, response.result.levels),
+      ),
+      true,
+    )
+    for (const key of Object.keys(response.result.levels)) {
+      assert.equal(typeof response.result.levels[key], 'string')
+    }
```
Comment (mvadari) [inline test/integration/requests/fee.ts:81]: ditto

File: test/integration/requests/serverInfo.ts [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+    assert.equal(response.status, expected.status)
+    assert.equal(response.type, expected.type)
+    assert.equal(
+      _.every(
+        Object.keys(expected.result.info),
+        // eslint-disable-next-line @typescript-eslint/unbound-method -- has verifies if the object actually has the key.
+        _.partial(_.has, response.result.info),
+      ),
+      true,
+    )
```
Comment (mvadari) [inline test/integration/requests/serverInfo.ts:63]: ditto

File: test/integration/utils.ts [hunk 15]
```diff
@@ -91,3 +92,7 @@ export async function testTransaction(
   await ledgerAccept(client)
   await verifySubmittedTransaction(client, signedTx as Transaction)
 }
+
+export function getMasterPublicKey(): string {
+  return keypairs.deriveKeypair(masterSecret).publicKey
+}
```
Comment (mvadari) [inline test/integration/utils.ts:98]: This doesn't look like it's used anywhere.

File: test/integration/requests/fee.ts (hunk 12 above)
Comment (mukulljangid) [reply test/integration/requests/fee.ts:68]: first part checks that the received response indeed has all the keys as the expected one. 
Second part checks for types of those

File: test/integration/utils.ts (hunk 15 above)
Comment (mukulljangid) [reply test/integration/utils.ts:98]: Ill remove!

File: test/integration/utils.ts [hunk 16]
```diff
@@ -1,6 +1,7 @@
 import { assert } from 'chai'
 import _ from 'lodash'
 import { decode } from 'ripple-binary-codec'
+import keypairs from 'ripple-keypairs'
```
Comment (mvadari) [inline test/integration/utils.ts:4]:
```suggestion
import { deriveKeypair } from 'ripple-keypairs'
```

File: test/integration/requests/fee.ts (hunk 12 above)
Comment (mvadari) [reply test/integration/requests/fee.ts:68]: You should be able to just directly compare the two objects - the numbers don't really change  (except for probably `ledger_current_index`)

File: test/integration/requests/fee.ts [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+        },
+        max_queue_size: '20000',
+      },
+      status: 'success',
+      type: 'response',
+    }
+    assert.equal(response.status, expected.status)
+    assert.equal(response.type, expected.type)
+
+    assert.deepEqual(
```
Comment (mvadari) [inline test/integration/requests/fee.ts:51]: check type on `ledger_current_index`

File: test/integration/requests/serverInfo.ts [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+      'uptime',
+      'seq',
+      'complete_ledgers',
+      'hostid',
+      'load',
+      'state_accounting',
+      'pubkey_node',
+      'server_state_duration_us',
+      'validated_ledger.hash',
+      'validated_ledger.seq',
```
Comment (mvadari) [inline test/integration/requests/serverInfo.ts:100]: ditto

File: test/integration/requests/serverState.ts [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+      'complete_ledgers',
+      'load',
+      'state_accounting',
+      'pubkey_node',
+      'time',
+      'uptime',
+      'server_state_duration_us',
+      'validated_ledger.hash',
+      'validated_ledger.seq',
+      'validated_ledger.close_time',
```
Comment (mvadari) [inline test/integration/requests/serverState.ts:80]: ditto

**Review by mvadari — COMMENTED** [review]
Why isn't `manifest` being tested?

**Comment by mukulljangid** [comment]
> Why isn't `manifest` being tested?

Works only on mainnet!

**Comment by natenichols** [comment]
>  Why isn't manifest being tested?

Manifest is a command that gets the identity of another validator based on its `master_key` or `ephemeral_key`. Since this is a standalone node, and thus does not communicate with any other validators, there are no manifests to retrieve.

File: test/integration/requests/serverState.ts [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+        state: {
+          build_version: '1.7.3',
+          complete_ledgers: '2563-2932',
+          io_latency_ms: 1,
+          jq_trans_overflow: '0',
+          last_close: {
+            converge_time: 100,
+            proposers: 0,
+          },
+          load: [Object],
```
Comment (mvadari) [inline test/integration/requests/serverState.ts:36]: ?

File: test/integration/requests/serverState.ts [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+          load_factor_fee_reference: 256,
+          load_factor_server: 256,
+          peer_disconnects: '0',
+          peer_disconnects_resources: '0',
+          peers: 0,
+          pubkey_node: 'n9K6DaaReKkCjb9sEfXh5xP3BV9JisrJ9biKB3CSSFXancBnv5cW',
+          pubkey_validator: 'none',
+          server_state: 'full',
+          server_state_duration_us: '8752487389',
+          state_accounting: [Object],
```
Comment (mvadari) [inline test/integration/requests/serverState.ts:50]: ?

**Review by mvadari — APPROVED** [review]
a couple of notes, please fix before resolving

**Pull request merged**